### PR TITLE
BUGFIX: TNT-1442 Theming and color palette ids for tiger

### PR DESCRIPTION
**Pull request by oleg-morozov**
JIRA: TNT-1442

---

Supported PR commands:

| Command                                                 | Description                                                |
| ------------------------------------------------------- | ---------------------------------------------------------- |
| `ok to test`                                            | Re-run standard checks                                     |
| `extended check sonar`                                  | SonarQube tests                                            |
| `extended test - backstop`                              | BackstopJS tests                                           |
| **E2E Cypress tests commands - TIGER**                  |                                                            |
| `extended test - tiger-cypress - isolated <testName>`   | Run isolated tests running against recorded Tiger backend. |
| `extended test - tiger-cypress - record <testName>`     | Create a new recording for isolated Tiger tests.           |
| `extended test - tiger-cypress - integrated <testName>` | Run integrated tests against live backend                  |
| **E2E Cypress tests commands - BEAR**                   |                                                            |
| `extended test - cypress - isolated <testName>`         | Run isolated tests running against recorded Bear backend.  |
| `extended test - cypress - record <testName>`           | Create a new recording for isolated Bear tests.            |
| `extended test - cypress - integrated <testName>`       | Run integrated tests against live backend                  |

`<testName>` in cypress commands is used to filter specfiles. Example, to run record with BEAR backend

-   Against `dashboard.spec.ts` and `drilling.spec.ts`, execute command `extended test - cypress - record dashboard,drilling`
-   Against all specfiles, execute command `extended test - cypress - record` or `extended test - cypress - record *`

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `extended test - backstop` passes
-   [ ] `extended test - tiger-cypress - record` to record new mapping files (Tiger BE)
-   [ ] `extended test - cypress - record` to record new mapping files (Bear BE)
-   [ ] `extended test - tiger-cypress - isolated` passes
-   [ ] `extended test - cypress - isolated` passes
-   [ ] `extended test - tiger-cypress - integrated` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/13133/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/14871/ : FAILURE in 6m 25s

**Comment by oleg-morozov** [comment]
extended test - backstop

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/13134/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-pull-request-dispatcher-pipeline/594/ : FAILURE in 7m 04s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/13135/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/14873/ : SUCCESS in 27m 20s

**Comment by oleg-morozov** [comment]
extended test - backstop

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/2102/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/13136/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-pull-request-dispatcher-pipeline/595/ : SUCCESS in 26m 50s

**Comment by oleg-morozov** [comment]
extended test - cypress - isolated *

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/13152/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-pull-request-dispatcher-pipeline/609/ : SUCCESS in 12m 11s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/13155/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-pull-request-dispatcher-pipeline/611/ : SUCCESS in 9m 20s

File: libs/sdk-backend-tiger/src/backend/organization/settings.ts [hunk 1]
```diff
@@ -83,4 +99,16 @@ export class OrganizationSettingsService
             }),
         );
     }
+
+    protected async deleteSettingByType(type: TigerSettingsType): Promise<void> {
+        const {
+            data: { data },
+        } = await this.getSettingByType(type);
+        const settingId = data.find((setting) => setting.attributes?.type === type)?.id;
```
Comment (xMort) [inline libs/sdk-backend-tiger/src/backend/organization/settings.ts:107]: I know that this is unlikely but imho this should rather use `filter` than `find` and then delete all the settings that were found.

File: libs/sdk-backend-tiger/src/backend/user/settings.ts [hunk 2]
```diff
@@ -86,4 +86,20 @@ export class TigerUserSettingsService
             });
         });
     }
+
+    protected async deleteSettingByType(type: TigerSettingsType): Promise<any> {
+        const {
+            data: { data },
+        } = await this.getSettingByType(type);
+        const settingId = data.find((setting) => setting.attributes?.type === type)?.id;
+        if (settingId) {
```
Comment (xMort) [inline libs/sdk-backend-tiger/src/backend/user/settings.ts:95]: Consider deleting all the matching settings, not just the first one, please.

File: libs/sdk-backend-tiger/src/backend/workspace/settings/index.ts [hunk 3]
```diff
@@ -88,6 +88,21 @@ export class TigerWorkspaceSettings
             }),
         );
     }
+
+    protected async deleteSettingByType(type: TigerSettingsType): Promise<any> {
+        const {
+            data: { data },
+        } = await this.getSettingByType(type);
+        const settingId = data.find((setting) => setting.attributes?.type === type)?.id;
+        if (settingId) {
```
Comment (xMort) [inline libs/sdk-backend-tiger/src/backend/workspace/settings/index.ts:97]: Consider deleting all the matching settings, not just the first one, please.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/13163/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/14878/ : SUCCESS in 17m 13s

**Review by xMort — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR3712

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/2673/ : SUCCESS in 11m 40s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/554/ : FAILURE in 1m 14s (non-voting)
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/14909/ : SUCCESS in 17m 28s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tiger-isolated-zuul-pipeline/165/ : SUCCESS in 13m 16s (non-voting)
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul-pipeline/201/ : SUCCESS in 12m 06s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/3828/ : SUCCESS in 16s

**Pull request merged**